### PR TITLE
chore(docs): Pre-encoded unicode characters can't be used in paths

### PR DESCRIPTION
**Pull request by Auspicus**
## Description

I ran into this weird behavior when creating a multilingual site where the paths were coming pre-encoded from the CMS. I was expecting that Gatsby would recognise pre-encoded paths and make the necessary adjustments so that I wouldn't have to do handle anything but I later found that paths MUST be encoded or pages break and bad things happen :o

I think this could be changed internally so that we don't have to rely on the developer to pass the correct value and instead handle the encoding inside Gatsby. For the meantime however, a docs update should suffice so that developers don't have to dig into the internals to understand whether or not unicode is supported and how to use it.

TLDR; Made a few changes to the API docs to mention how paths should not be encoded when containing unicode characters.

### Documentation

- [x] Add a section into the `path` field description in `createPage` API
- [x] Add a section in the routing docs explaining that path is not expected to be encoded

## Related Issues

Related to #16765

### CODE REVIEW

**Comment by Auspicus** [comment]
@LekoArts I went and updated the docs to note the way that `createPage` is currently expecting the `path` param to not be encoded based on that old ticket: #36325. Is this the right direction or should there be a change to the API to support encoded paths?

**Review by LekoArts — APPROVED** [review]
Thanks!
